### PR TITLE
CR-1099860 xbutil validate shows unclear message about skipped/unsupported tests

### PR DESCRIPTION
**Pull request by rchane**
#### Problem solved by the commit
https://jira.xilinx.com/browse/CR-1099860
#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
As delineated, test descriptions were previously printed for black-box tests before running them. As such, skipped black-box tests due to lack of support would leave a hanging test title before moving onto the next test. This PR aims to eliminate such hanging titles and to only provide this information when `--verbose` is enabled, cleaning up the output.
#### How problem was solved, alternative solutions (if any) and why they were rejected
Thanks to previous efforts, there is now a `--supported` flag to query if a specific test case is supported on a device. This is utilized to determine whether or not to proceed with running the test/printing its output. In addition to this, testing code was refactored in order to help with readability and organization.
#### Risks (if any) associated the changes in the commit
This affects what the user will see when running `xbutil validate` as described in the CR. Users will need to use `--verbose` to see skipped black-box tests. There may also be merge conflicts due to the new structure of how one adds new black-box test case handlers (any additional checks outside of `-s` for black-box tests should be handled in `check_test_supported`).
#### What has been tested and how, request additional testing if necessary
Testing has been done on xsjsagarw50's devices.
```
rchane@xsjsagarw50:/proj/rdi/staff/rchane/XRT$ xbutil validate -d 18:00
Starting validation for 1 devices

Validate Device           : [0000:18:00.1]
    Platform              : xilinx_u30_gen3x4_base_2
    SC Version            : 
    Platform ID           : 6E7C97F7-949F-5106-3075-A77784C30A4B
-------------------------------------------------------------------------------
Test 1 [0000:18:00.1]     : pcie-link 
    Test Status           : [PASSED]
-------------------------------------------------------------------------------
```
More tests pass by...
```
Test 5 [0000:18:00.1]     : iops 
    Details               : IOPS: 116512 (verify)
    Test Status           : [PASSED]
-------------------------------------------------------------------------------
Test 6 [0000:18:00.1]     : mem-bw 
    Details               : Throughput (Type: DDR) (Bank count: 1) : 8840.93MB/s
    Test Status           : [PASSED]
-------------------------------------------------------------------------------
Validation completed, but with warnings. Please run the command '--verbose' option for more details
```
Skipped black-box tests were not printed, but can be seen with `--verbose`, e.g.
```
-------------------------------------------------------------------------------
Test 10 [0000:18:00.1]    : hostmem-bw 

    Description           : Run 'bandwidth kernel' when host memory is enabled
    Xclbin                : /opt/xilinx/firmware/u30/gen3x4/base/test
    Testcase              : /opt/xilinx/xrt/test/hostmemory.exe
    Details               : Test is not supported on current device.
    Test Status           : [SKIPPED]
-------------------------------------------------------------------------------
```

### CODE REVIEW

**Comment by rchane** [comment]
Would it be possible for me to find a list of which tests are supported on which devices? While I did a lot of sanity checking while refactoring code, I would like to be further certain.

**Comment by dayeh-xilinx** [comment]
retest this please.

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 1]
```diff
@@ -296,91 +286,141 @@ runTestCase( const std::shared_ptr<xrt_core::device>& _dev, const std::string& p
...
+        exit_code = XBU::runScript("sh", xrtTestCasePath, args, "Checking Test Support", "Test Support Check Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
+        break;
+      case test_mode::no_json:
+        if (_ptTest.get<std::string>("file").find(".exe") != std::string::npos)
+          exit_code = XBU::runScript("", xrtTestCasePath, args, "Running Test", "Test Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
+        else
+          exit_code = XBU::runScript("python", xrtTestCasePath, args, "Running Test", "Test Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
+        break;
+      default:
+        return false; //Mode was not specified correctly.
```
Comment (dbenusov-xilinx) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:332]:
```suggestion
        throw std::runtime_exception("Invalid test mode specified"); //Mode was not specified correctly.
```
That way the message will be logged and we can easily find the issue!

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 2]
```diff
@@ -296,91 +286,141 @@ runTestCase( const std::shared_ptr<xrt_core::device>& _dev, const std::string& p
...
-    boost::filesystem::path xrt_path(xrtTestCasePath);
-    if (!boost::filesystem::exists(xrt_path)) {
-      logger(_ptTest, "Error", boost::str(boost::format("Failed to find %s") % xrtTestCasePath));
-      logger(_ptTest, "Error", "Please check if the platform package is installed correctly");
-      _ptTest.put("status", test_token_failed);
-      return;
-    }
-    // log testcase path for debugging purposes
-    logger(_ptTest, "Testcase", xrtTestCasePath);
+}
```
Comment (dbenusov-xilinx) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:354]: There is no return if the code makes it to the end of the function. I think it should return false if we get here.

**Review by dbenusov-xilinx — CHANGES_REQUESTED** [review]
Great job! This code segment looks much cleaner.

**Comment by rchane** [comment]
> Great job! This code segment looks much cleaner.

Thank you! I went ahead and implemented the changes you requested.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Pull request closed without merging**